### PR TITLE
webinstall: Offer kraftkit installation

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

The `webinstall` tool is used to offer an easy, guided, and automated way to install `kraftkit`.

An earlier version is already available and this build on top of it to ensure that the script is maintainable and covers all the corner cases.
It will be used to deploy the script.

For detailed information, see the commit comments.

### CODE REVIEW

**Review by StefanJum — APPROVED** [review]
All good, thanks @craciunoiuc.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Comment by craciunoiuc** [comment]
@nderjung 😉 😉

File: tools/webinstall/main.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+const (
+	DefaultScriptPath = "/"
+	DefaultLatestPath = "/latest.txt"
+)
+
+type Webinstall struct {
+	Freq  time.Duration `long:"freq"  short:"F" usage:"The frequency (in hours) to check for updates" default:"24h"`
+	Port  int           `long:"port"  short:"P" usage:"The port to serve the script" default:"8080"`
+	Token string        `long:"token" short:"T" usage:"The GitHub token for querying tags" default:""`
```
Comment (nderjung) [inline tools/webinstall/main.go]:
```diff
-       Freq  time.Duration `long:"freq"  short:"F" usage:"The frequency (in hours) to check for updates" default:"24h"`
-       Port  int           `long:"port"  short:"P" usage:"The port to serve the script" default:"8080"`
-       Token string        `long:"token" short:"T" usage:"The GitHub token for querying tags" default:""`
+       Freq  time.Duration `long:"freq"  short:"F" usage:"The frequency (in hours) to check for updates" env:"WEBINSTALL_FREQ" default:"24h"`
+       Port  int           `long:"port"  short:"P" usage:"The port to serve the script" env:"WEBINSTALL_PORT" default:"8080"`
+       Token string        `long:"token" short:"T" usage:"The GitHub token for querying tags" env:"WEBINSTALL_TOKEN" default:""`
```

File: tools/webinstall/go.mod [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module github.com/unikraft/kraftkit/webinstall
+
+go 1.18
+
+require (
+	github.com/rancher/wrangler v1.1.1
+	github.com/spf13/cobra v1.6.1
+	kraftkit.sh v0.4.1
```
Comment (nderjung) [inline tools/webinstall/go.mod]:
```diff
-       kraftkit.sh v0.4.1
+       kraftkit.sh v0.4.1-117-g39fdf8e
```

File: tools/webinstall/go.mod [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+require (
+	github.com/AlecAivazis/survey/v2 v2.3.6 // indirect
+	github.com/aymanbagabas/go-osc52 v1.2.1 // indirect
+	github.com/briandowns/spinner v1.21.0 // indirect
```
Comment (nderjung) [inline tools/webinstall/go.mod:14]:
```diff
  	github.com/briandowns/spinner v1.21.0 // indirect
+       github.com/charmbracelet/lipgloss v0.6.0 // indirect
+       github.com/cli/go-gh v1.1.0 // indirect
```

File: tools/webinstall/go.sum [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+github.com/AlecAivazis/survey/v2 v2.3.6/go.mod h1:4AuI9b7RjAR+G7v9+C4YSlX/YL3K3cWNXgWXOhllqvI=
+github.com/Netflix/go-expect v0.0.0-20220104043353-73e0943537d2/go.mod h1:HBCaDeC1lPdgDeDbhX8XFpy1jqjK0IBG8W5K+xYqA0w=
+github.com/aymanbagabas/go-osc52 v1.2.1 h1:q2sWUyDcozPLcLabEMd+a+7Ea2DitxZVN9hTxab9L4E=
+github.com/aymanbagabas/go-osc52 v1.2.1/go.mod h1:zT8H+Rk4VSabYN90pWyugflM3ZhpTZNC7cASDfUCdT4=
+github.com/briandowns/spinner v1.21.0 h1:2lVBzf3iZ3cT/ulVXljc4BzlL3yTWZDzsGsamI7si+A=
+github.com/briandowns/spinner v1.21.0/go.mod h1:TcwZHb7Wb6vn/+bcVv1UXEzaA4pLS7yznHlkY/HzH44=
+github.com/cli/safeexec v1.0.1 h1:e/C79PbXF4yYTN/wauC4tviMxEV13BwljGj0N9j+N00=
+github.com/cli/safeexec v1.0.1/go.mod h1:Z/D4tTN8Vs5gXYHDCbaM1S/anmEDnJb1iW0+EJ5zx3Q=
+github.com/cpuguy83/go-md2man/v2 v2.0.2/go.mod h1:tgQtvFlXSQOSOSIRvRPT7W67SCa46tRHOmNcaadrF8o=
+github.com/creack/pty v1.1.17/go.mod h1:MOBLtS5ELjhRRrroQr9kyvTxUAFNvYEK993ew/Vr4O4=
```
Comment (nderjung) [inline tools/webinstall/go.sum]:
```diff
 github.com/briandowns/spinner v1.21.0/go.mod h1:TcwZHb7Wb6vn/+bcVv1UXEzaA4pLS7yznHlkY/HzH44=
+github.com/charmbracelet/lipgloss v0.6.0 h1:1StyZB9vBSOyuZxQUcUwGr17JmojPNm87inij9N3wJY=
+github.com/charmbracelet/lipgloss v0.6.0/go.mod h1:tHh2wr34xcHjC2HCXIlGSG1jaDF0S0atAUvBMP6Ppuk=
+github.com/cli/go-gh v1.1.0 h1:E6uVflEDYz+m/UDxMbkx0jaIdKbrTXC+m1MfnkygiYU=
+github.com/cli/go-gh v1.1.0/go.mod h1:2TxNyy9B/tD1rpL5xqHyHRdy0+jg86zaJePEekkxEZc=
```

Comment (nderjung) [reply tools/webinstall/go.sum]: (etc)

File: tools/webinstall/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ENV CGO_ENABLED=0
+
+COPY . /go/src/webinstall
+
+RUN set -xe; \
+    go build -tags static -a -ldflags='-s -w' -ldflags '-extldflags "-static"' .
+
+FROM scratch AS prod
+
+COPY --from=base /go/src/webinstall/webinstall /webinstall
```
Comment (nderjung) [inline tools/webinstall/Dockerfile]:
```diff
+FROM alpine:3.17.2 as ca-certificates
+
+RUN apk add -U --no-cache ca-certificates
+
 FROM scratch AS prod
 
+COPY --from=ca-certificates /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
 COPY --from=base /go/src/webinstall/webinstall /webinstall
 ```

**Review by nderjung — APPROVED** [review]
Awesome! Thanks :) 

Approved-by: Alexander Jung <alex@unikraft.io>

**Pull request merged**